docs/source/app-dev/services.rst

```diff
@@ -0,0 +1,159 @@
+
+Command service
+===============
+
+Use the **command service** when you want to submit a command and wait for it to be executed. This service is similar to the command submission service, but also receives completions and waits until it knows whether or not the submitted command has completed. It returns the completion status of the command execution.
+
+You can use either the command or command submission services to submit commands to effect a ledger change. The command service is useful for simple applications, as it handles a basic form of coordination between command submission and completion, correlating submissions with completions, and returning a success or failure status. This allow simple applications to be completely stateless, and alleviates the need for them to track command submissions.
+
+Reading from the ledger
+***********************
+
+Transaction service
+===================
+
+Use the **transaction service** to listen to changes in the ledger state, reported via a stream of transaction events.
+
+Transaction events detail the changes on transaction boundaries - each event denotes a transaction on the ledger, and contains all the update events (create, exercise, archive of contracts) that had an effect in that transaction.
+
+Transaction events contain a :ref:`transactionId <com.digitalasset.ledger.api.v1.Transaction.transaction_id>` (assigned by the server), the ``workflowId``, the ``commandId``, and the events in the transaction.
+
+Transaction events are the primary mechanism by which an application will do its work. Event-driven applications can use them to generate new commands, and state-driven applications will use them to update their state model, by e.g. creating data that represents created contracts.
+
+The transaction service can be initiated to read events from an arbitrary point on the ledger. This is important when starting or restarting and application, and works in conjunction with the `active contract service <#active-contract-service>`__.
+
+Transaction and transaction trees
+---------------------------------
+
+``TransactionService`` offers several different subscriptions. The most commonly used is ``GetTransactions``. If you need more details, you can use ``GetTransactionTrees`` instead, which returns transactions as flattened trees, represented as a map of event IDs to events and a list of root event IDs.
+
+.. _verbosity:
+
+Verbosity
+---------
+
+The service works in a non-verbose mode by default, which means that some identifiers are omitted:
+
+- Record IDs
+- Record field labels
+- Variant IDs
+
+You can get these included in requests related to Transactions by setting the ``verbose`` field in message ``GetTransactionsRequest`` or ``GetActiveContractsRequest`` to ``true``.
+
+Active contract service
+=======================
+
+Use the **active contract service** to obtain a party-specific view of all the contracts recently active on the ledger.
```

Comment:
I don't like the phrasing "recently active". Maybe just say "current" with a footnote or say "a recent snapshot of active contracts".